Request fails when using a streaming response in spring boot apps

Hello,
I am experimenting with logbook and wanted to try to see how it behaves with streams and asynchronous responses.
I have created a small example to show the issue [here](https://github.com/sgandon/test-zalando-stream)
When I set the `logbook.write.level=INFO` and call the endpoint this just breaks the request completely

The stack trace is quite big but we can see the following messages.

> Caused by: java.lang.IllegalStateException: getOutputStream() has already been called for this response
> java.nio.BufferOverflowException: null
> and some NPEs.

the proposal above creates a nice NPE
```
Caused by: java.lang.NullPointerException: null
	at java.util.Arrays.stream(Arrays.java:5004)
	at org.zalando.logbook.MediaTypeQuery.compile(MediaTypeQuery.java:19)
	at org.zalando.logbook.Conditions.contentType(Conditions.java:47)
	at org.talend.components.service.rest.configuration.ApiVersionConfig.setupLogBook(ApiVersionConfig.java:39)
```
I have also tried with the content type empty ("") and I have another exception
```
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:1967)
	at org.zalando.logbook.MediaTypeQuery.compile(MediaTypeQuery.java:30)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:479)
	at org.zalando.logbook.MediaTypeQuery.compile(MediaTypeQuery.java:21)
	at org.zalando.logbook.Conditions.contentType(Conditions.java:47)
	at org.talend.components.service.rest.configuration.ApiVersionConfig.setupLogBook(ApiVersionConfig.java:39)
```

--- a/logbook-core/src/main/java/org/zalando/logbook/Conditions.java
+++ b/logbook-core/src/main/java/org/zalando/logbook/Conditions.java
@@ -50,6 +50,10 @@ public final class Conditions {
                 query.test(message.getContentType());
     }
 
+    public static <T extends BaseHttpMessage> Predicate<T> withoutContentType() {
+        return message -> message.getContentType() == null;
+    }
+
     public static <T extends BaseHttpMessage> Predicate<T> header(final String key, final String value) {
         return message ->
                 message.getHeaders().getOrDefault(key, emptyList()).contains(value);

--- a/logbook-core/src/test/java/org/zalando/logbook/ConditionsTest.java
+++ b/logbook-core/src/test/java/org/zalando/logbook/ConditionsTest.java
@@ -11,10 +11,11 @@ import static org.zalando.logbook.Conditions.contentType;
 import static org.zalando.logbook.Conditions.exclude;
 import static org.zalando.logbook.Conditions.header;
 import static org.zalando.logbook.Conditions.requestTo;
+import static org.zalando.logbook.Conditions.withoutContentType;
 
 public final class ConditionsTest {
 
-    private final RawHttpRequest request = MockRawHttpRequest.create()
+    private final MockRawHttpRequest request = MockRawHttpRequest.create()
             .withHeaders(MockHeaders.of("X-Secret", "true"))
             .withContentType("text/plain");
 
@@ -87,6 +88,20 @@ public final class ConditionsTest {
     @Test
     void contentTypeShouldNotMatch() {
         final Predicate<BaseHttpMessage> unit = contentType("application/json");
+
+        assertThat(unit.test(request), is(false));
+    }
+
+    @Test
+    void withoutContentTypeShouldMatch() {
+        final Predicate<BaseHttpMessage> unit = withoutContentType();
+
+        assertThat(unit.test(request.withContentType(null)), is(true));
+    }
+
+    @Test
+    void withoutContentTypeShouldNotMatch() {
+        final Predicate<BaseHttpMessage> unit = withoutContentType();
 
         assertThat(unit.test(request), is(false));
     }

--- a/logbook-core/src/test/java/org/zalando/logbook/MediaTypeQueryTest.java
+++ b/logbook-core/src/test/java/org/zalando/logbook/MediaTypeQueryTest.java
@@ -14,6 +14,8 @@ public final class MediaTypeQueryTest {
     void shouldMatchAllMatch() {
         deny("*/*", null);
         deny("*/*", "");
+        deny("text/*", "application/json");
+        deny("text/plain", "application/json");
         allow("*/*", "text/plain");
         allow("text/*", "text/plain");
         allow("text/plain", "text/plain");

